### PR TITLE
Add new check - CF - CKV_AWS_166 (AmazonMQ Public Access)

### PR DESCRIPTION
**Pull request by shubh-cs29**
Hello,

Implementing check for point in time recovery (Backup) for DynamoDB Global Table in CloudFormation.

CloudFormation Docs:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-amazonmq-broker.html#cfn-amazonmq-broker-users

Note:
AmazonMQ Broker's PubliclyAccessible value should be false/no.

License
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/cloudformation/checks/resource/aws/AmazonMQBrokerPublicAccess.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
+
+class AmazonMQBrokerPublicAccess(BaseResourceCheck):
+
+    def __init__(self):
+        name = "Ensure Amazon MQ Broker should not have public access"
+        id = "CKV_AWS_166"
```
Comment (gruebel) [inline checkov/cloudformation/checks/resource/aws/AmazonMQBrokerPublicAccess.py:8]: Can you change the id to `CKV_AWS_69` then it correspond to the Terraform check

File: checkov/cloudformation/checks/resource/aws/AmazonMQBrokerPublicAccess.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        supported_resources = ['AWS::AmazonMQ::Broker']
+        categories = [CheckCategories.GENERAL_SECURITY]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
```
Comment (gruebel) [inline checkov/cloudformation/checks/resource/aws/AmazonMQBrokerPublicAccess.py:13]: I think in general you can make the check much simpler, by subclassing `BaseResourceValueCheck` like here
https://github.com/bridgecrewio/checkov/blob/1e8a4aa2bae94bf571fe73f88338075f91056d59/checkov/cloudformation/checks/resource/aws/RDSPubliclyAccessible.py#L5-L19

File: tests/cloudformation/checks/resource/aws/example_AmazonMQBrokerPublicAccess/AmazonMQBrokerPublicAccess-FAILED-1.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      EngineType: RABBITMQ
+      EngineVersion: 3.8.6
+      HostInstanceType: mq.t3.micro
+      Logs:
+        General: true
+      MaintenanceWindowStartTime:
+        DayOfWeek: Monday
+        TimeOfDay: '22:45'
+        TimeZone: America/Los_Angeles
+      PubliclyAccessible: Yes
```
Comment (gruebel) [inline tests/cloudformation/checks/resource/aws/example_AmazonMQBrokerPublicAccess/AmazonMQBrokerPublicAccess-FAILED-1.yaml:27]: just out of curiosity, is Yes and No valid values for a boolean value in CloudFormation?

**Review by gruebel — CHANGES_REQUESTED** [review]

File: tests/cloudformation/checks/resource/aws/example_AmazonMQBrokerPublicAccess/AmazonMQBrokerPublicAccess-FAILED-1.yaml (hunk 3 above)
Comment (shubham-cldcvr) [reply tests/cloudformation/checks/resource/aws/example_AmazonMQBrokerPublicAccess/AmazonMQBrokerPublicAccess-FAILED-1.yaml:27]: @gruebel yes/no/true/false/on/off all are valid boolean values in CloudFormation

Comment (shubh-cs29) [reply tests/cloudformation/checks/resource/aws/example_AmazonMQBrokerPublicAccess/AmazonMQBrokerPublicAccess-FAILED-1.yaml:27]: @gruebel yes/no/true/false/on/off are all these valid boolean values in CloudFormation. I have tested these.

File: checkov/cloudformation/checks/resource/aws/AmazonMQBrokerPublicAccess.py (hunk 1 above)
Comment (shubh-cs29) [reply checkov/cloudformation/checks/resource/aws/AmazonMQBrokerPublicAccess.py]: @gruebel changed the id to **CKV_AWS_69**

File: checkov/cloudformation/checks/resource/aws/AmazonMQBrokerPublicAccess.py (hunk 2 above)
Comment (shubh-cs29) [reply checkov/cloudformation/checks/resource/aws/AmazonMQBrokerPublicAccess.py]: @gruebel  made the changes in code as per your suggestions.

**Review by nimrodkor — APPROVED** [review]
Nice! Thanks @shubh-cs29 !
And thank you as well for the guidance @gruebel !

**Pull request merged**